lib/systemFlows.js

```diff
@@ -1,22 +1,41 @@
 'use strict'
 
-var getSystemInfo = require('./system').getSystemInfo
+var system = require('./system')
 var api = require('./util').api
 var flowTimeoutConditionTimestamps = {}
+var checkActionSystemNewFirmwareTimeout
 const language = Homey.manager('i18n').getLanguage()
+const checkTimeOut = 15 * 60 * 1000
 
 exports.init = function () {
   Homey.manager('flow').trigger('systemInitiated', null, null)
   Homey.manager('flow').on('condition.systemCloudConnected', onConditiononSystemCloudConnected)
-  Homey.manager('flow').on('condition.systemFlowTimeout', onConditiononSystemFlowTimeout)
-  Homey.manager('flow').on('condition.systemFlowTimeout.units.autocomplete', onConditionSystemFlowTimeoutUnitsAutocomplete)
   Homey.manager('flow').on('condition.systemDeviceLastUpdate', onConditiononSystemDeviceLastUpdate)
   Homey.manager('flow').on('condition.systemDeviceLastUpdate.deviceCapability.autocomplete', onConditiononSystemDeviceLastUpdateDeviceCapabilityAutocomplete)
   Homey.manager('flow').on('condition.systemDeviceLastUpdate.units.autocomplete', onConditiononSystemDeviceLastUpdateUnitsAutocomplete)
+  Homey.manager('flow').on('condition.systemFlowTimeout', onConditiononSystemFlowTimeout)
+  Homey.manager('flow').on('condition.systemFlowTimeout.units.autocomplete', onConditionSystemFlowTimeoutUnitsAutocomplete)
+  Homey.manager('flow').on('condition.systemNewFirmware', onConditionSystemNewFirmware)
+
+  checkActionSystemNewFirmwareTimeout = setInterval(checkActionSystemNewFirmware, checkTimeOut)
+  checkActionSystemNewFirmware()
+}
+
+function checkActionSystemNewFirmware () {
+  var tokens = {}
+  system.getUpdateInfo().then(function (updateInfo) {
```

Comment:
use `Promise.all([system.getSystemInfo(), system.getUpdateInfo()])` for parallel requests